torch_xla/csrc/translator.cpp

```diff
@@ -192,6 +193,476 @@ XlaTranslationResult XlaTranslator::BuildComputation(
+  cctx->AddValueOp(node_outputs[2], grads.grad_bias);
+}
+
+void TranslateSum(const torch::jit::Node* node, ComputationContext* cctx,
+                  xla::PrecisionConfig::Precision /*conv_precision*/,
+                  xla::XlaBuilder* /*b*/) {
+  XLA_CHECK_GE(node->inputs().size(), 1);
+  xla::XlaOp xla_output = BuildSum(node, cctx->OpForInput(node, 0));
+  cctx->AddNodeOp(node, xla_output);
+}
+
+void TranslateNllLoss(const torch::jit::Node* node, ComputationContext* cctx,
+                      xla::PrecisionConfig::Precision /*conv_precision*/,
+                      xla::XlaBuilder* /*b*/) {
+  XLA_CHECK_EQ(node->inputs().size(), 5);
+  xla::XlaOp xla_output =
+      BuildNllLoss(node, cctx->OpForInput(node, 0), cctx->OpForInput(node, 1));
+  cctx->AddNodeOp(node, xla_output);
+}
+
+void TranslateNllLossBackward(
+    const torch::jit::Node* node, ComputationContext* cctx,
+    xla::PrecisionConfig::Precision /*conv_precision*/,
+    xla::XlaBuilder* /*b*/) {
+  XLA_CHECK_EQ(node->inputs().size(), 7);
+  xla::XlaOp xla_output = BuildNllLossBackward(node, cctx->OpForInput(node, 1),
+                                               cctx->OpForInput(node, 2));
+  cctx->AddNodeOp(node, xla_output);
+}
+
+void TranslateSize(const torch::jit::Node* node, ComputationContext* cctx,
+                   xla::PrecisionConfig::Precision /*conv_precision*/,
+                   xla::XlaBuilder* /*b*/) {
+  XLA_CHECK_EQ(node->inputs().size(), 1);
+  std::vector<xla::int64> size_op_result;
+  xla::XlaOp xla_output =
+      BuildSize(node, cctx->OpForInput(node, 0), &size_op_result);
+  cctx->AddSizeOpResult(node->output(0), size_op_result);
+  cctx->AddNodeOp(node, xla_output);
+}
+
+void TranslateConstant(const torch::jit::Node* node, ComputationContext* cctx,
+                       xla::PrecisionConfig::Precision /*conv_precision*/,
+                       xla::XlaBuilder* b) {
+  cctx->AddNodeOp(node, GetConstantOp(b, node));
+}
+
+void TranslateUndefined(const torch::jit::Node* node, ComputationContext* cctx,
+                        xla::PrecisionConfig::Precision /*conv_precision*/,
+                        xla::XlaBuilder* /*b*/) {
+  cctx->AddUndefinedInput(ComputationContext::OutputId(node));
+}
+
+void TranslateSumToSize(const torch::jit::Node* node, ComputationContext* cctx,
+                        xla::PrecisionConfig::Precision /*conv_precision*/,
+                        xla::XlaBuilder* /*b*/) {
+  XLA_CHECK_EQ(node->inputs().size(), 2);
+  xla::XlaOp xla_output =
+      BuildSumToSize(node, cctx->OpForInput(node, 0), cctx->GetSizeOpValues());
+  cctx->AddNodeOp(node, xla_output);
+}
+
+void TranslateNop(const torch::jit::Node* /*node*/,
+                  ComputationContext* /*cctx*/,
+                  xla::PrecisionConfig::Precision /*conv_precision*/,
+                  xla::XlaBuilder* /*b*/) {}
+
+}  // namespace
+
+std::unordered_map<torch::jit::NodeKind, XlaTranslator::TranslationHandler>
```

Comment:
Do not keep destructible global data. Create and store a global pointer instead.